sdk/js/src/api/http.js

```diff
@@ -112,7 +113,16 @@ class Http {
       return response
     } catch (err) {
       if ('response' in err && err.response && 'data' in err.response) {
-        throw err.response.data
+        const error = cloneDeep(err.response.data)
+        error.request_details = {
+          url: get(err, 'response.config.url'),
+          method: get(err, 'response.config.method'),
+          params: get(err, 'response.config.params'),
+          base_url: get(err, 'response.config.baseURL'),
+          stack_component: parsedComponent,
+        }
```

Comment:
> Not too happy about this.

Me neither, but its the simplest solution that can be changed at any moment, does not require any api changes in the console/sdk and uses functionality that we already have in the sdk. Also, this provides additional debug info (http method, full url and target stack component can be helpful).

> IMO, a better option would be to change the output for device CRUD functions in the SDK so that when an error occurs, they return an object of errors keyed by component ID. 

Then I would expect device CRUD methods to return successful results in a similar shape.
```
{
  is: {... device fields stored in is ...},
  as: {... device fields stored in as ...},
  ...
}
```

We can adjust this by changing signatures of device crud methods, for example
```
async create(applicationId, device, mask = Marshaler.fieldMaskFromPatch(device))
// can be changed to:
async create(applicationId, device, mask = Marshaler.fieldMaskFromPatch(device), mergeResult = true)
// or
async create(applicationId, device, mask = Marshaler.fieldMaskFromPatch(device), { mergeResult = true })
// or
async create(applicationId, device, { mask = Marshaler.fieldMaskFromPatch(device), mergeResult = true })
```

Basically when `mergeResult` is `true` then both merged result and first error is returned, otherwise we return results split by component, see above. This will require some changes to end device redux primitives.

I can add utility functions to `lib/errors/utils.js` to get the target component and use those instead of directly accessing `request_details` in the device-add wizard. So later we can switch to other implementation (we can check for failed component by error message or anything else)

Reply:
Indeed, I was thinking of a solution like that. Can you put this into a new issue and add a `TODO:` here?

Note that when manipulating the error object, these things will be part of the displayed error details in the Console. Meaning that they won't represent the backend error 1:1 anymore. This might confuse others when trying to debug.